xds/internal/balancer/clusterresolver/configbuilder.go

```diff
@@ -234,12 +248,97 @@ func dedupSortedIntSlice(a []int) []int {
 	return a[:i+1]
 }
 
+// priorityLocalitiesToClusterImpl takes a list of localities (with the same
+// priority), and generates a cluster impl policy config, and a list of
+// addresses.
+func priorityLocalitiesToClusterImpl(localities []xdsclient.Locality, priorityName string, mechanism DiscoveryMechanism, drops []clusterimpl.DropConfig, xdsLBPolicy *internalserviceconfig.BalancerConfig) (*clusterimpl.LBConfig, []resolver.Address, error) {
+	var addrs []resolver.Address
+	clusterImplCfg := &clusterimpl.LBConfig{
+		Cluster:                 mechanism.Cluster,
+		EDSServiceName:          mechanism.EDSServiceName,
+		LoadReportingServerName: mechanism.LoadReportingServerName,
+		MaxConcurrentRequests:   mechanism.MaxConcurrentRequests,
+		DropCategories:          drops,
+		// ChildPolicy is not set. Will be set based on xdsLBPolicy
+	}
+
+	if xdsLBPolicy == nil || xdsLBPolicy.Name == rrName {
+		// If lb policy is ROUND_ROBIN:
+		// - locality-picking policy is weighted_target
+		// - endpoint-picking policy is round_robin
+		logger.Infof("xds lb policy is %q, building config with weighted_target + round_robin", rrName)
+		var wtConfig *weightedtarget.LBConfig
+		wtConfig, addrs = localitiesToWeightedTarget(
+			localities, priorityName,
+			// Child of weighted_target is hardcoded to round_robin.
+			&internalserviceconfig.BalancerConfig{Name: roundrobin.Name},
+		)
```

Comment:
Nit: We don't usually break up lines this way. It was a little hard to read initially. Could we avoid this?

Reply:
Done